extensions/amp-mowplayer/0.1/amp-mowplayer.js

```diff
@@ -255,20 +263,19 @@ class AmpMowplayer extends AMP.BaseElement {
   }
 
   /**
-   * Sends 'listening' message to the Mowplayer iframe to listen for events.
+   * Check player is visible or not based on breakpoint and send message to player
    * @private
    */
-  listenToFrame_() {
-    if (!this.iframe_) {
-      return;
-    }
+  onVisibilityObserver_() {
+    const worker = () => {
+      const {intersectionRatio} = this.element.getIntersectionChangeEntry();
+      const visible = intersectionRatio > 0.5 ? true : false;
+      this.sendMessage_('visibility_observer_visibility', {'visible': visible});
+    };
 
-    this.sendCommand_('listening', [
-      'amp',
-      window.location.href,
-      window.location.origin,
-      true,
-    ]);
+    this.viewport_.onChanged(worker);
```

Comment:
This `onChanged` will register an additional event listener **every time** this message is received. Please remove.